server/src/main/java/org/opensearch/cluster/routing/WeightedRoutingService.java

```diff
@@ -106,6 +108,39 @@ private boolean checkIfSameWeightsInMetadata(
         return newWeightedRoutingMetadata.getWeightedRouting().equals(oldWeightedRoutingMetadata.getWeightedRouting());
     }
 
+    public void deleteWeightedRoutingMetadata(
+        final ClusterDeleteWeightedRoutingRequest request,
+        final ActionListener<ClusterDeleteWeightedRoutingResponse> listener
+    ) {
+        clusterService.submitStateUpdateTask("delete_weighted_routing", new ClusterStateUpdateTask(Priority.URGENT) {
+            @Override
+            public ClusterState execute(ClusterState currentState) {
+                logger.info("Deleting weighted routing metadata from the cluster state");
+                Metadata.Builder mdBuilder = Metadata.builder(currentState.metadata());
+                mdBuilder.removeCustom(WeightedRoutingMetadata.TYPE);
+                return ClusterState.builder(currentState).metadata(mdBuilder).build();
+            }
+
+            @Override
+            public void onFailure(String source, Exception e) {
+                logger.error(
+                    () -> new ParameterizedMessage(
+                        "failed to remove weighted routing metadata from cluster state with an exception [{}]",
+                        e
+                    )
+                );
```

Comment:
if there's no other parameter we can keep it like - `logger.error(msg, e)`

Reply:
addressed